infrastructure/app.py

```diff
@@ -1,17 +1,393 @@
+                    subnet_type=aws_ec2.SubnetType.PRIVATE_ISOLATED,
+                    cidr_mask=24,
+                ),
+            ],
+            nat_gateways=app_config.nat_gateway_count,
+        )
+
+        self.vpc.add_interface_endpoint(
+            "SecretsManagerEndpoint",
+            service=aws_ec2.InterfaceVpcEndpointAwsService.SECRETS_MANAGER,
+        )
+
+        self.vpc.add_interface_endpoint(
+            "CloudWatchEndpoint",
+            service=aws_ec2.InterfaceVpcEndpointAwsService.CLOUDWATCH_LOGS,
+        )
+
+        self.vpc.add_gateway_endpoint(
+            "S3", service=aws_ec2.GatewayVpcEndpointAwsService.S3
+        )
+
+        self.export_value(
+            self.vpc.select_subnets(subnet_type=aws_ec2.SubnetType.PUBLIC)
+            .subnets[0]
+            .subnet_id
+        )
+        self.export_value(
+            self.vpc.select_subnets(subnet_type=aws_ec2.SubnetType.PUBLIC)
+            .subnets[1]
+            .subnet_id
+        )
+
+
+class eoAPIStack(Stack):
+    def __init__(
+        self,
+        scope: Construct,
+        vpc: aws_ec2.Vpc,
+        id: str,
+        app_config: AppConfig,
+        **kwargs,
+    ) -> None:
+        super().__init__(
+            scope,
+            id=id,
+            tags=app_config.tags,
+            **kwargs,
+        )
+
+        #######################################################################
+        # PG database
+        pgstac_db = PgStacDatabase(
+            self,
+            "pgstac-db",
+            vpc=vpc,
+            engine=aws_rds.DatabaseInstanceEngine.postgres(
+                version=aws_rds.PostgresEngineVersion.VER_14
+            ),
+            vpc_subnets=aws_ec2.SubnetSelection(
+                subnet_type=(
+                    aws_ec2.SubnetType.PUBLIC
+                    if app_config.public_db_subnet
+                    else aws_ec2.SubnetType.PRIVATE_ISOLATED
+                )
+            ),
+            allocated_storage=app_config.db_allocated_storage,
+            instance_type=aws_ec2.InstanceType(app_config.db_instance_type),
+            removal_policy=RemovalPolicy.DESTROY,
+        )
+        pgstac_db.db.connections.allow_default_port_from_any_ipv4()
+
+        #######################################################################
+        # STAC API service
+        stac_api_lambda = PgStacApiLambda(
+            self,
+            "stac-api",
+            api_env={
+                "NAME": app_config.build_service_name("stac"),
+                "description": f"{app_config.stage} STAC API",
+            },
+            db=pgstac_db.db,
+            db_secret=pgstac_db.pgstac_secret,
+            # If the db is not in the public subnet then we need to put
+            # the lambda within the VPC
+            vpc=vpc if not app_config.public_db_subnet else None,
+            subnet_selection=aws_ec2.SubnetSelection(
+                subnet_type=aws_ec2.SubnetType.PRIVATE_WITH_EGRESS
+            )
```

Comment:
if the database is in a private subnet we need to put the lambdas in the private subnet as well